### ISSUE TITLE
Investigate using `nucleo` (and others) for fuzzy matching

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/zed#7586]
### Check for existing issues

- [X] Completed

### Describe the feature

Our fuzzy matcher algorithm is the best:

Zed:

![SCR-20240208-mjgw](https://github.com/zed-industries/zed/assets/19867440/70619ba5-2900-4b7e-9e8a-1a10a2ea82e3)

VS Code:
![SCR-20240208-mjue](https://github.com/zed-industries/zed/assets/19867440/a0cd0643-c803-4b1d-8154-68e91fc6c7e2)

Investigate the quality of matches that can be had by implementing [`nucleo`](https://github.com/helix-editor/nucleo), or any other well-tested library. Check the accuracy of

- file finder results
- command palette results
- search buffer symbols results
- etc.

This is a research issue. If the results don't pan out, we will stick to our custom matcher and try to improve it over time.

**Comment by JosephTLyons** [comment]
https://github.com/zed-industries/zed/issues/4315

**Comment by WeetHet** [comment]
Yeah, I think using nucleo would be the best, as now there are multiple issues with the fuzzy search in the editor, including autocompletion and file finding

**Comment by Blankeos** [comment]
<img width="1107" alt="Image" src="https://github.com/user-attachments/assets/d2aa36b7-fe94-42a2-9ef6-8c36f004ad1b" />

The amount of times this happens ^ is kind of annoying haha, if nucleo fixes this, it would be swell. Does Neovim and VSCode use nucleo btw?

**Comment by yara-blue** [comment]
Partially implemented, further implementation is tracked in https://github.com/zed-industries/zed/issues/55872. 

If anyone wants to help further this effort by adding a few more pickers let us know in the tracking issue!

@Nereuxofficial this turned out to be a group effort! Your about to owe cookies to a group of awesome contributors :)

**Comment by Nereuxofficial** [comment]
> Partially implemented, further implementation is tracked in https://github.com/zed-industries/zed/issues/55872. 
> 
> If anyone wants to help further this effort by adding a few more pickers let us know in the tracking issue!
> 
> @Nereuxofficial this turned out to be a group effort! Your about to owe cookies to a group of awesome contributors :)

Awesome! Nice to see this implemented, can't wait to see this in stable.

From the tracking issue I was able to gather a list of contributors to this feature

@11happy @feitreim @yara-blue @MrSubidubi  Thanks for working on this!
Could you please reach out to me via email at nereux@tuta.io so you can claim your cookies?
Also if I've missed anyone please notify me.

**Comment by MrSubidubi** [comment]
@Nereuxofficial Sincerely appreciate you keeping your word here but if anyone on the team should get those cookies, I think Yara deserves these - she kept pushing this forward and I just quickly jumped in helping out with one panic that was introduced. Thus, she should also get the cookies in my opinion 🙁

**Comment by Nereuxofficial** [comment]
> [@Nereuxofficial](https://github.com/Nereuxofficial) Sincerely appreciate you keeping your word here but if anyone on the team should get those cookies, I think Yara deserves these - she kept pushing this forward and I just quickly jumped in helping out with one panic that was introduced. Thus, she should also get the cookies in my opinion 🙁

Sounds good!  @yara-blue reach out to nereux@tuta.io if you want to claim your cookies here!
